### DISCUSSION TITLE
Pointnet2_classification.py each eposch training time

### DISCUSSION THREAD

**Discussion by utkarsh0902311047** [pyg-team/pytorch_geometric#8130] [Q&A] [answered]
Hello 
I am trying the Pointnet2_classification.py example and its each epoch training takes approx 2min and 10 sec. 
I am using cuda 10.1 with GPU NVIDIA T1000 8gb.

Is it too slow or its the appropriate training time?

**Comment by rusty1s** [comment]
A single epoch + testing takes around 17 seconds for me. Which PyG and PyTorch version are you using?

**Reply by utkarsh0902311047** [reply]
Cuda:10.1
Pytorch: 1.8.1
Pyg: 2.2.0

**Reply by rusty1s** [reply]
I think there was a bug in `BatchNorm` for older PyTorch versions. You could try to remove the `BatchNorm` and see if that speeds things up.

**Reply by utkarsh0902311047** [reply]
self.mlp = MLP([1024, 512, 256, 10], dropout=0.5, norm=None)
Removing BatchNorm is done by setting norm=None in Pointnet2_classification.py line 56, or is there any other way?
The above code line already has norm=None

**Reply by rusty1s** [reply, marked as answer]
You would need to specify `norm=None` in every initialization of `MLP`, e.g., in `SAModule`.

**Reply by utkarsh0902311047** [reply]
Ok now it is taking around 30sec to train every epoch+testing. thank you so much.
What should be the appropriate pytorch, PyG, and Cuda versions to make it work without any bugs such that also it can be run in editable mode?

**Reply by rusty1s** [reply]
I think it was fixed around PyTorch 1.10, but I am not totally sure. I suggest to always ensure you are running on either one of the two most recent PyTorch versions.
